app/js/components/orderEntry/OrderEntryPage.jsx

```diff
@@ -2,44 +2,49 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
 import PatientDashboard from '../patientDashboard';
-import { withRouter } from 'react-router-dom';
 import SearchAndAddOrder from './SearchAndAddOrder';
-import { fetchInpatientCareSetting, fetchOutpatientCareSetting } from '../../actions/careSetting';
+import fetchPatientCareSetting from '../../actions/careSetting';
+import imageLoader from '../../../img/loading.gif';
 
-class OrderEntryPage extends React.Component {
+export class OrderEntryPage extends React.Component {
   componentDidMount() {
-    this.props.fetchInpatientCareSetting();
-    this.props.fetchOutpatientCareSetting();
+    this.props.fetchPatientCareSetting();
   }
   render() {
+    if (!(this.props.outpatientCareSetting || this.props.inpatientCareSetting)) {
```

Comment:
Correct me if i'm wrong, but his algorithm implies that if outpatientCareSetting and/or inpatientCareSetting has a valid value (1), it should show the loader...
```
!(1 + 0) = !1 = 0
!(1 + 1) = !1 = 0
!(0 + 0) = !0 = 1
```
I'm thinking it should be 
```
(!this.props.outpatientCareSetting || !this.props.inpatientCareSetting)
or 
!(this.props.outpatientCareSetting && this.props.inpatientCareSetting) - distributive law of logic
```

Reply:
Will look into this.

Reply:
Had an initial implementation using `(!this.props.outpatientCareSetting || !this.props.inpatientCareSetting)`, while refactoring the code further to include both cases with a single negation, did not replace the `||`.
Thanks for pointing this out.

Reply:
Implementation updated now.